### PR TITLE
Issue with creating DB during GRANT 

### PR DESCRIPTION
**Pull request by przemas75**
https://github.com/owncloud/core/issues/28010
It worked with 9.1.6 version.
It is based on https://github.com/owncloud/core/pull/25485/commits/1080afe6eed3a7e2d70bae13d49a8468c735585f which was not merged to version 9 of owncloud.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@PVince81 agree with this backport?

File: lib/private/Setup/MySQL.php [hunk 1]
```diff
@@ -60,12 +60,20 @@ private function createDatabase($connection) {
 			//we can't use OC_BD functions here because we need to connect as the administrative user.
 			$query = "CREATE DATABASE IF NOT EXISTS `$name` CHARACTER SET utf8 COLLATE utf8_bin;";
 			$connection->executeUpdate($query);
-
+			} catch (\Exception $ex) {
+			$this->logger->error('Database creation failed: {error}', [
```
Comment (PVince81) [inline lib/private/Setup/MySQL.php]: please indent with tabs

File: lib/private/Setup/MySQL.php [hunk 2]
```diff
@@ -60,12 +60,20 @@ private function createDatabase($connection) {
...
+				'app' => 'mysql.setup',
```
Comment (PVince81) [inline lib/private/Setup/MySQL.php]: app should be "core"

Comment (PVince81) [inline lib/private/Setup/MySQL.php]: never mind, seems someone else set it to something else before...

**Comment by PVince81** [comment]
backport to stable10 sounds fine

**Comment by PVince81** [comment]
@przemas75 can you update the PR according to my comments ?

**Comment by przemas75** [comment]
@PVince81  I can see tabs in the changes, not sure what shall i correct.

**Comment by PVince81** [comment]
one additional tab is missing on that line

File: lib/private/Setup/MySQL.php (hunk 1 above)
Comment (przemas75) [inline lib/private/Setup/MySQL.php]: Is that proper indent?

**Comment by PVince81** [comment]
Looks good now 👍

**Pull request merged**